src/test/java/org/openrewrite/staticanalysis/AnnotateNullableMethodsTest.java

```diff
@@ -334,4 +335,42 @@ public String getString() {
           )
         );
     }
+
+    @Test
+    void nestedType() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              package a;
+              public class B {
+                  public static class C {}
+              }
+              """,
+            SourceSpec::skip
+          ),
+          //language=java
+          java(
+            """
+              import a.B;
+              public class Foo {
+                  public B.C bar() {
+                      return null;
+                  }
+              }
+              """,
+            """
+              import a.B;
+              import org.jspecify.annotations.Nullable;
+
+              public class Foo {
+
+                  public  B.@Nullable C bar() {
```

Comment:
Not quite happy with this extra newline & double space yet, which are part of moving the annotation after it was added.

Reply:
Looks like this should be fixed upstream, so we'll leave this here for now.
https://github.com/openrewrite/rewrite-static-analysis/blob/a779afabaa0d12c77df16a384520e3a64bc762bd/src/main/java/org/openrewrite/staticanalysis/java/MoveFieldAnnotationToType.java#L80-L84